### PR TITLE
VM Update September 2022

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

**Comment by talregev** [comment]
This time for real?

**Comment by Cheney-W** [comment]
The VS bug of colmap and graphicsmagick has not been fixed yet.

**Comment by BillyONeal** [comment]
> This time for real?

Hopefully!

> The VS bug of colmap and graphicsmagick has not been fixed yet.

Yeah I was just giving them a shot.

**Comment by BillyONeal** [comment]
REGRESSION: colmap:x64-windows failed with BUILD_FAILED. If expected, add colmap:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: colmap:x64-windows-static failed with BUILD_FAILED. If expected, add colmap:x64-windows-static=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: colmap:x64-windows-static-md failed with BUILD_FAILED. If expected, add colmap:x64-windows-static-md=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: colmap:x86-windows failed with BUILD_FAILED. If expected, add colmap:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: graphicsmagick:x64-windows failed with BUILD_FAILED. If expected, add graphicsmagick:x64-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: graphicsmagick:x64-windows-static failed with BUILD_FAILED. If expected, add graphicsmagick:x64-windows-static=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: graphicsmagick:x64-windows-static-md failed with BUILD_FAILED. If expected, add graphicsmagick:x64-windows-static-md=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: graphicsmagick:x86-windows failed with BUILD_FAILED. If expected, add graphicsmagick:x86-windows=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: mfl:x64-windows-static failed with BUILD_FAILED. If expected, add mfl:x64-windows-static=fail to C:\a\2\s\scripts\azure-pipelines/../ci.baseline.txt.
REGRESSION: mfl:x64-windows-static-md failed with BUILD_FAILED. If expected, add mfl:x64-windows-static-md=fail to C:\a\1\s\scripts\azure-pipelines/../ci.baseline.txt.

These were all skipped before; I put their skips back.

PASSING, REMOVE FROM FAIL LIST: jemalloc:x64-osx (/Users/vagrant/Data/work/2/s/scripts/azure-pipelines/../ci.baseline.txt).

:)

REGRESSION: duktape:x64-linux failed with BUILD_FAILED. If expected, add duktape:x64-linux=fail to /agent/_work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

Traceback (most recent call last):
  File "tools/configure.py", line 29, in <module>
    import yaml
ImportError: No module named yaml

I think this needs the venv treatment, investigating.

REGRESSION: shogun:x64-linux failed with BUILD_FAILED. If expected, add shogun:x64-linux=fail to /agent/_work/2/s/scripts/azure-pipelines/../ci.baseline.txt.

FAILED: src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o 
/usr/bin/c++ -DH5_BUILT_AS_STATIC_LIB -I/mnt/vcpkg-ci/buildtrees/shogun/src/b686d6e1cc-76ef87b3ad.clean/src -I/mnt/vcpkg-ci/buildtrees/shogun/x64-linux-dbg/src -I/mnt/vcpkg-ci/installed/x64-linux/include/libxml2 -isystem /mnt/vcpkg-ci/installed/x64-linux/include/eigen3 -isystem /mnt/vcpkg-ci/installed/x64-linux/include -Wall -Wno-unused-parameter -Wformat -Wformat-security -Wparentheses -Wshadow -Wno-unknown-pragmas -Wno-deprecated -fPIC -g -fPIC -gsplit-dwarf -std=c++11 -MD -MT src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o -MF src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o.d -o src/shogun/CMakeFiles/libshogun.dir/classifier/svm/NewtonSVM.cpp.o -c /mnt/vcpkg-ci/buildtrees/shogun/src/b686d6e1cc-76ef87b3ad.clean/src/shogun/classifier/svm/NewtonSVM.cpp
In file included from /mnt/vcpkg-ci/buildtrees/shogun/src/b686d6e1cc-76ef87b3ad.clean/src/shogun/classifier/svm/NewtonSVM.cpp:20:
/mnt/vcpkg-ci/buildtrees/shogun/src/b686d6e1cc-76ef87b3ad.clean/src/shogun/mathematics/lapack.h:25:10: fatal error: mkl_cblas.h: No such file or directory
   25 | #include <mkl_cblas.h>
      |          ^~~~~~~~~~~~~
compilation terminated.

I think this is an optional dependency detection, investigating.

File: ports/shogun/portfile.cmake [hunk 1]
```diff
@@ -44,6 +44,7 @@ vcpkg_configure_cmake(
         -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=TRUE
         -DCMAKE_DISABLE_FIND_PACKAGE_CURL=TRUE
         -DCMAKE_DISABLE_FIND_PACKAGE_OpenMP=TRUE
+        -DBLA_VENDOR=OpenBLAS
```
Comment (Neumann-A) [inline ports/shogun/portfile.cmake:47]: that is not for this port to decide. see #24327

Comment (BillyONeal) [reply ports/shogun/portfile.cmake:47]: That makes sense; unfortunately without it I can't install `intel-mkl` on the Linux VMs. @JackBoosY are you OK with not taking that this time? It was added for https://github.com/microsoft/vcpkg/pull/15233

Comment (JackBoosY) [reply ports/shogun/portfile.cmake:47]: Okay.

Comment (BillyONeal) [reply ports/shogun/portfile.cmake:47]: Reverting this change for now then. Thanks for the feedback

**Comment by dg0yt** [comment]
I would like to remind you that `libtool-bin` could be added to the linux image, cf. #26923.

**Comment by BillyONeal** [comment]
> I would like to remind you that `libtool-bin` could be added to the linux image, cf. #26923.

Not really a reminder, I didn't see the other one. If I have to mint the image again I will include that, if not I'll get it next month. (Although 'next month' will be closer to 2-3 weeks because we're delayed in landing this one :( )

**Comment by BillyONeal** [comment]
OK the linux failure in https://dev.azure.com/vcpkg/public/_build/results?buildId=78582&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172 is exactly the same as the current fleet's build https://dev.azure.com/vcpkg/public/_build/results?buildId=78538&view=results , so we're good to go.

**Pull request merged**